nilearn/plotting/surf_plotting.py

```diff
@@ -501,19 +500,20 @@ def _plot_surf_matplotlib(coords, faces, surf_map=None, bg_map=None,
             face_colors[kept_indices] = cmap(surf_map_faces[kept_indices])
 
         if colorbar:
-            ticks = _get_ticks_matplotlib(vmin, vmax, cbar_tick_format)
-            our_cmap, norm = _get_cmap_matplotlib(cmap, vmin, vmax, threshold)
-            bounds = np.linspace(vmin, vmax, our_cmap.N)
+            cbar_vmin = cbar_vmin if cbar_vmin is not None else vmin
+            cbar_vmax = cbar_vmax if cbar_vmax is not None else vmax
+            ticks = _get_ticks_matplotlib(cbar_vmin, cbar_vmax, cbar_tick_format)
+            our_cmap, norm = _get_cmap_matplotlib(cmap, cbar_vmin, cbar_vmax, threshold)
```

Comment:
Thanks a lot @RaphaelMeudec ! 

the 2d plots are looking great.

Here I think the vmin and vmax are for the color**map** itself, see colorbar of the surface plot you show in the discussion. Setting the colorbar boundaries as you do below should be enough to take care of the colorbar.

I'll have more time for a complete review later this week!

```suggestion
            our_cmap, norm = _get_cmap_matplotlib(cmap, vmin, vmax, threshold)
```